app/lib/features/invite_members/widgets/invite_code_ui.dart

```diff
@@ -0,0 +1,142 @@
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _InviteCodeUIState();
+}
+
+class _InviteCodeUIState extends ConsumerState<InviteCodeUI> {
+  @override
+  Widget build(BuildContext context) {
+    var inviteCode = ref
+        .watch(inviteCodeForSelectRoomOnly(widget.roomId))
+        .valueOrNull
+        ?.token();
+    if (inviteCode != null) {
+      return Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          DottedBorderWidget(
+            child: Row(
+              children: [
+                Expanded(
+                  child: Text(
+                    inviteCode,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+                IconButton(
+                  onPressed: () {
+                    Clipboard.setData(
+                      ClipboardData(text: inviteCode),
+                    );
+                  },
+                  icon: const Icon(Icons.copy),
+                ),
+              ],
+            ),
+          ),
+          const SizedBox(height: 10),
+          Align(
+            alignment: Alignment.centerRight,
+            child: ActerInlineTextButton(
+              onPressed: () async {
+                final token =
+                    await ref.read(superInviteTokenProvider(inviteCode).future);
+                if (!context.mounted) return;
+                context.pushNamed(
+                  Routes.actionCreateSuperInvite.name,
+                  extra: token,
+                );
+              },
+              child: Text(
+                L10n.of(context).manage,
+              ),
+            ),
+          ),
+          const SizedBox(height: 10),
+          ActerPrimaryActionButton(
+            onPressed: () => context.pushNamed(
+              Routes.shareInviteCode.name,
+              queryParameters: {
+                'inviteCode': inviteCode,
+                'roomId': widget.roomId,
+              },
+            ),
+            child: Text(L10n.of(context).share),
+          ),
+        ],
+      );
+    }
+    return ActerPrimaryActionButton(
+      onPressed: () => generateNewInviteCode(context, ref),
+      child: Text(L10n.of(context).generateInviteCode),
+    );
+  }
+
+  Future<void> generateNewInviteCode(
+    BuildContext context,
+    WidgetRef ref,
+  ) async {
+    try {
+      EasyLoading.show(status: L10n.of(context).generateInviteCode);
+      final displayName =
+          await ref.read(roomDisplayNameProvider(widget.roomId).future);
+      String prefix =
+          (displayName?.replaceAll(RegExp(r'[^A-Za-z]'), '').toLowerCase() ??
+              '');
+
+      final rng = Random();
+
+      int end = 5;
+      if (prefix.isEmpty) {
+        end = 8;
+      } else if (prefix.length > 8) {
+        prefix = prefix.substring(0, 8);
+        end = 3;
+      } else if (prefix.length > 4) {
+        end = 3;
+      }
```

Comment:
How about putting this logic on new code generation provider?

Reply:
I opted for keeping the code generator dumb for now as given that it can have multiple rooms entered and doesn't (at this point) actually mess with any refs or something like it, I'd like to see more usage patterns before generalizing this. But yeah, if we see this taking place at more than this location we should probably move it out.